pkg/asset/agent/manifests/infraenv.go

```diff
@@ -131,5 +142,11 @@ func (i *InfraEnv) finish() error {
 		return errors.New("missing configuration or manifest file")
 	}
 
+	// Throw an error if CpuArchitecture isn't x86_64, aarch64, or ""
+	switch i.Config.Spec.CpuArchitecture {
+	case arch.RpmArch(types.ArchitectureAMD64), arch.RpmArch(types.ArchitectureARM64), "":
```

Comment:
@jeffdyoung I have a question: IIUC accepting the empty value `""` here will force any subsequent consumer of this field to check for a valid default (I noticed at least two dfferent ones, one in `ignition.go` and another one in `baseiso.go`). Wouldn't be easier (and less prone to error) to set the default directly here and just once for all, if the field is empty?

Reply:
@andfasano Initially I did what you are saying, but I was getting a few unit test errors in https://github.com/openshift/installer/blob/master/pkg/asset/agent/manifests/infraenv_test.go
```
--- FAIL: TestInfraEnv_Generate (0.00s)
    --- FAIL: TestInfraEnv_Generate/valid_configuration (0.00s)
        infraenv_test.go:132: 
            	Error Trace:	/go/src/github.com/openshift/installer/pkg/asset/agent/manifests/infraenv_test.go:132
            	Error:      	Received unexpected error:
            	            	Config.Spec.CpuArchitecture  is not supported 
 .
 .
 .
      --- FAIL: TestInfraEnv_Generate/proxy_valid_configuration (0.00s)
        infraenv_test.go:132: 
            	Error Trace:	/go/src/github.com/openshift/installer/pkg/asset/agent/manifests/infraenv_test.go:132
            	Error:      	Received unexpected error:
            	            	Config.Spec.CpuArchitecture  is not supported 
```
Which made me think that there might be other applications that would expect to use `""` as an architecture. ([Hypershift](https://hypershift-docs.netlify.app/how-to/agent/create-agent-cluster/) may be one of those?)
Basically I did it this way so I wouldn't break other things 🤞

Reply:
That said, I'm happy to make the changes if you all feel its safe to do: @zaneb @andfasano

Reply:
@jeffdyoung in general looks good to me, and I don't want to block on this specific point that could be potentially improved in a subsequent PR. Let's see if @zaneb may have any specific comment on that

Reply:
I can imagine a scenario where install-config has `amd64` and InfraEnv being `multi`, but we aren't there yet.

Reply:
@andfasano @zaneb we are good to resolve, and address this issue in the future when we start addressing issues with the `multi` payload?